### PR TITLE
evidently 0.4.39 :snowflake:

### PR DESCRIPTION
**Pull request by boldorider4**
evidently 0.4.39 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5359]
- dev_url:        https://github.com/evidentlyai/evidently/tree/v0.4.39
- conda_forge:    https://github.com/conda-forge/evidently-feedstock/blob/main/recipe/meta.yaml
- pypi:           https://pypi.org/project/evidently/0.4.39
- pypi inspector: https://inspector.pypi.io/project/evidently/0.4.39

### Explanation of changes:

- new version number

[PKG-5359]: https://anaconda.atlassian.net/browse/PKG-5359?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,39 +1,83 @@
...
   number: 0
   # umap-learn isn't available on s390x
   skip: True  # [py<37 or s390x]
-  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  # missing dynaconf for py312
+  skip: True  # [py>311]
+  script:
+    - {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+    # removing experimental features that are dependent on missing chromadb
+    - rm -rf $SP_DIR/evidently/experimental
```
Comment (cbouss) [inline recipe/meta.yaml]: I wouldn't remove it. 
chromadb is optional: https://github.com/evidentlyai/evidently/blob/v0.4.39/setup.py#L106
It is expected to fail if not install. If a user wants to use this experimental feature, they could always use chromadb from another source.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -44,11 +88,16 @@ test:
...
+    - pytorch
+    - pillow>=10.3.0
+    - setuptools <=65.5.1 # [py<312]
+    - setuptools ==68.2.2 # [py>=312]
     # To fix `from numba.np.ufunc import _internal SystemError: initialization of _internal failed without raising an exception`
     - numba >=0.57.0
   commands:
-    - pip check
-    - pytest tests -vv
+    # skipping pip check since it detects uuid6 0.0.0.dev0 when uuid6==2024.7.10 is used
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:114]: The package `uuid6`  has a wrong `METADATA` file and should be patched to fix the issue.
It's caused by a specific env variable required for retrieving a version, see https://github.com/oittaa/uuid6-python/blob/2024.07.10/setup.py#L15-L19.
I think it's fine to skip it here but could you please create a Jira ticket to rebuild `uuid6`?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,39 +1,83 @@
...
     - python
-    - pyyaml >=5.1
-    - requests >=2.19.0
-    - scikit-learn >=0.24.0
-    - scipy >=1.5.4
+    - plotly >=5.10.0
     - statsmodels >=0.12.2
-    - umap-learn >=0.5.3
+    - scikit-learn >=1.0.1
+    - pandas >=1.3.5
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:39]: Should be `pandas[parquet]>=1.3.5`. That means we need `pyarrow >=10.0.1` too, see https://github.com/pandas-dev/pandas/blob/v2.2.2/pyproject.toml#L71C12-L71C27

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,39 +1,83 @@
...
+    - cryptography >=43.0.1
+    # transformers would be optional but it is included in BERTScore_descriptor.py
+    - transformers >=4.39.3
+  # from https://github.com/evidentlyai/evidently/blob/cd6cbfba3f0c0ebdb7494e53941d954272f451fa/requirements.min.txt#L30
+  run_constrained:
+    - openai >=1.16.2
+    - evaluate >=0.4.1
+    - sentence-transformers >=2.7.0
+    - chromadb >=0.4.0
+    - blas=*=openblas  # [osx and x86_64]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:77]: Should `pytorch` and `accelerate` be here also? See https://github.com/evidentlyai/evidently/blob/v0.4.39/setup.py#L104 and https://github.com/huggingface/transformers/blob/v4.45.2/setup.py#L280

Comment (skupr-anaconda) [reply recipe/meta.yaml:77]: And these optional dependencies https://github.com/evidentlyai/evidently/blob/v0.4.39/setup.py#L108C20-L111

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,39 +1,83 @@
...
   number: 0
   # umap-learn isn't available on s390x
   skip: True  # [py<37 or s390x]
-  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  # missing dynaconf for py312
+  skip: True  # [py>311]
+  script:
+    - {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+    # removing experimental features that are dependent on missing chromadb
+    - rm -rf $SP_DIR/evidently/experimental
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:24]:
```suggestion
  entry_points:
    - evidently=evidently.cli:app

```
https://github.com/evidentlyai/evidently/blob/v0.4.39/setup.py#L115C40-L115C67

File: recipe/meta.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/meta.yaml]: So in this case I should patch out or remove some tests? Will do.

Comment (boldorider4) [reply recipe/meta.yaml]: I'm always a little skeptical of packages that declare a dependency as optional and then use it broadly around the code, where code may fail if the dependency is not installed. But if we want to reproduce upstream behavior to the letter, I'm just going to patch out the tests.

File: recipe/meta.yaml (hunk 2 above)
Comment (boldorider4) [reply recipe/meta.yaml:114]: ok.

Comment (boldorider4) [reply recipe/meta.yaml:114]: created here https://anaconda.atlassian.net/browse/PKG-6267

File: recipe/meta.yaml (hunk 4 above)
Comment (boldorider4) [reply recipe/meta.yaml:77]: I could add them, yes.

Comment (boldorider4) [reply recipe/meta.yaml:77]: actually there is no version constraint on `pytorch`. I added `accelerate` and a few more optional dependencies.

File: recipe/meta.yaml (hunk 3 above)
Comment (boldorider4) [reply recipe/meta.yaml:39]: I added `pyarrow >=10.0.1` to `run_constrained`.

File: recipe/meta.yaml (hunk 5 above)
Comment (boldorider4) [reply recipe/meta.yaml:24]: good catch!

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml:39]: It should be in run.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,39 +1,91 @@
...
+    - pandas >=1.3.5
+    - numpy >=1.22.0,<2.1
+    - nltk >=3.6.7
+    - scipy >=1.10.0
+    - requests >=2.32.0
+    - PyYAML >=5.4
+    - pydantic >=1.10.13
+    - litestar >=2.8.3
+    - typing_inspect >=0.9.0
+    - uvicorn >=0.22.0
```
Comment (cbouss) [inline recipe/meta.yaml:50]: It's asking for standard here: https://github.com/evidentlyai/evidently/blob/v0.4.39/setup.py#L66
```suggestion
    - uvicorn >=0.22.0
    - colorama >=0.4 # [win]
    - httptools >=0.5.0
    - python-dotenv >=0.13
    - uvloop >=0.14.0,!=0.15.0,!=0.15.1 # [unix]
    - watchfiles >=0.13
    - websockets >=10.4
```
https://inspector.pypi.io/project/uvicorn/0.30.6/packages/5a/01/5e637e7aa9dd031be5376b9fb749ec20b86f5a5b6a49b87fabd374d5fa9f/uvicorn-0.30.6.tar.gz/uvicorn-0.30.6/pyproject.toml#line.39

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,39 +1,91 @@
...
+    - transformers >=4.39.3
+    # because of transformers[torch]
+    - accelerate >=0.26.0
+    # because of [spark]
+    - pyspark >=3.4.0
+    # because of [fsspec]
+    - s3fs >=2024.9.0
+    - gcsfs >=2024.9.0
+    # because of pandas[parquet]
+    - pyarrow >=10.0.1
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
Should be in run.

File: recipe/meta.yaml [hunk 8]
```diff
@@ -44,11 +96,17 @@ test:
     - pip
     - pytest
     - ipython
+    - pytorch
+    - pillow>=10.3.0
+    - setuptools <=65.5.1 # [py<312]
+    - setuptools ==68.2.2 # [py>=312]
     # To fix `from numba.np.ufunc import _internal SystemError: initialization of _internal failed without raising an exception`
     - numba >=0.57.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
Pretty sure that's not required anymore.

File: recipe/meta.yaml (hunk 3 above)
Comment (boldorider4) [reply recipe/meta.yaml:39]: done.

File: recipe/meta.yaml (hunk 7 above)
Comment (boldorider4) [reply recipe/meta.yaml]: fixed

File: recipe/meta.yaml (hunk 6 above)
Comment (boldorider4) [reply recipe/meta.yaml:50]: didn't catch that!

Comment (boldorider4) [reply recipe/meta.yaml:50]: there's pyyaml too

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**